client-ui/src/app/components/lineage/lineage-details/schema-details/schema-details.component.ts

```diff
@@ -76,103 +95,55 @@ export class SchemaDetailsComponent implements AfterViewInit {
   }
 
-  private getType(attribute?: any): string {
-    return attribute._type
-  }
 
-  private getExpressions(attribute: any): Expression[] {
+  private getExpressions(detailsInfos: any, attributeList: any): Expression[] {
+    const properties = detailsInfos.operation.properties
     let expressions = []
-    switch (attribute._type) {
+    switch (properties.name) {
       case OperationType.Join:
-        // Build the join expression
-        const title = attribute.joinType
-        const values = [attribute.condition.text]
-        const expression = new Expression(title, values)
-        expressions.push(expression)
+        const joinExpression = new Expression(`${properties.joinType} Join On`, getText(properties.condition, attributeList), properties.condition)
+        expressions.push(joinExpression)
         break
       case OperationType.Projection:
-        // Build the transformations expressions
-        if (attribute.transformations) {
-          const title = "Transformations"
-          const values = new Array()
-          attribute.transformations.forEach(transformation => values.push(this.getText(transformation)))
-          const transformationExpression: Expression = new Expression(title, values)
-          expressions.push(transformationExpression)
+        if (properties.projectList) {
+          properties.projectList.forEach(projection =>
+            expressions.push(new Expression("Transformations", getText(projection[1], attributeList), projection[1]))
+          )
         }
-        // Build the dropped Attributes expressions
         let inputs = []
-        _.each(attribute.inputs, schemaIndex => inputs = _.concat(inputs, attribute.schemas[schemaIndex]))
-        const output = attribute.schemas[attribute.output]
+        _.each(detailsInfos.inputs, schemaIndex => inputs = _.concat(inputs, detailsInfos.schemas[schemaIndex]))
+        const output = detailsInfos.schemas[detailsInfos.output]
         const diff = _.differenceBy(inputs, output, 'name')
         if (diff.length > 0) {
-          const title = "Dropped Attributes"
-          const values = diff.map(item => item.name);
-          const droppedAttributes = new Expression(title, values)
-          expressions.push(droppedAttributes)
+          diff.map(item => expressions.push(new Expression("Dropped Attributes", item.name, null)))
         }
         break
-      //TODO : Implement the other expressions for the other types
+      case OperationType.Aggregate:
+        properties.aggregateExpressions.forEach(aggregate => expressions.push(new Expression("Aggregate", getText(aggregate[1], attributeList), aggregate[1])))
+        properties.groupingExpressions.forEach(grouping => expressions.push(new Expression("Grouping", getText(grouping[1], attributeList), grouping[1])))
+        break
+      case OperationType.LogicalRelation:
+        const inputSourceExpression = new Expression("inputSource", properties.inputSources[0], null)
```

Comment:
Why is `inputSource` expression? And why is only first one taken?

Reply:
fixed